src/commons/utils/JavaHelper.ts

```diff
@@ -148,3 +157,43 @@ export async function javaRun(javaCode: string, context: Context) {
       return { status: 'error' };
     });
 }
+
+export function visualizeJavaCseMachine({ context }: { context: ECE.Context }) {
+  try {
+    CseMachine.drawCse(context);
+  } catch (err) {
+    throw new Error('Java CSE machine is not enabled');
+  }
+}
+
+export async function runJavaCseMachine(code: string, targetStep: number, context: Context) {
+  const convertJavaErrorToJsError = (e: ECE.SourceError): SourceError => ({
+    type: ErrorType.RUNTIME,
+    severity: ErrorSeverity.ERROR,
+    location: {
+      start: {
+        line: 0,
+        column: 0
+      },
+      end: {
+        line: 0,
+        column: 0
+      }
```

Comment:
Is this not planned to be supported?

Reply:
At least not for this PR, errors thrown by ECE in `java-slang` currently do not have source node information.

Reply:
Noted, thanks! Perhaps we can leave a TODO comment

Reply:
Sure